src/state/swap/hooks.tsx

```diff
@@ -119,6 +120,18 @@ export function useDerivedSwapInfo(
     (isExactIn ? outputCurrency : inputCurrency) ?? undefined
   )
 
+  useEffect(() => {
+    if (trade.state === TradeState.INVALID) {
+      setPreviouslyInvalid(true)
+    } else if (trade.state !== TradeState.LOADING) {
+      setPreviouslyInvalid(false)
+    }
+  }, [trade.state])
+
+  if (trade.state == TradeState.LOADING && previouslyInvalid) {
+    trade.trade = undefined
```

Comment:
The mutation here feels wrong to me especially with the useMemo using trade below, I wonder if we shouldn't put this logic into `useBestTrade` behind a new option like `onlyWhenValid`? Otherwise, and either way, I think you'd want to do something like `let trade = useBestTrade` and then here do `trade = { ...trade, trade: undefined }`

Reply:
Fixed! Thanks for pointing out